slsim/lens.py

```diff
@@ -320,27 +277,36 @@ def deflector_redshift(self):
     def source_redshift(self):
         """
 
-        :return: source redshift
+        :return: list of source redshifts
         """
-        return self.source.redshift
+        source_redshift_list = []
+        for source in self.source:
+            source_redshift_list.append(source.redshift)
+        return source_redshift_list
 
     @property
     def external_convergence(self):
         """
 
         :return: external convergence
         """
-        _, _, kappa_ext = self.los_linear_distortions
-        return kappa_ext
+        kappa_list = []
+        for los_linear_dist in self.los_linear_distortions:
```

Comment:
I am not sure I understand the looping through here. There is only one line of sight per lens. The only thing that changes is the lensing efficiency part